execute/dataset.go

```diff
@@ -115,6 +129,14 @@ func (d *dataset) evalTriggers() (err error) {
 			// Skip the rest once we have encountered an error
 			return
 		}
+
+		select {
+		case <-d.ctx.Done():
```

Comment:
There is only one event in the select so I think we can just call `Err()` here. If the result is non-nill then return it. 
https://golang.org/pkg/context/#Context